File: mobile_deckz/lib/model/game_history/game_history.dart
import 'dart:core';

import 'package:mobile_deckz/model/game_history/game.dart';

class GameHistory {
  final List<Game> games;

  GameHistory({required this.games}) {
    games.sort((a, b) => b.timestamp.compareTo(a.timestamp));
  }

  factory GameHistory.fromJson(Map<String, dynamic> json) {
    List<Game> games = [];
    for (var game in json['games']) {
      games.add(Game.fromJson(game));
    }
    return GameHistory(games: games);
  }

  int getNumberOfGames() {
    return games.length;
  }

  int totalScore() {
    return games.map((game) => game.getScore()).reduce((a, b) => a + b);

  }
}

File: src/ulb/infof307/g01/model/gamehistory/GameHistory.java
package ulb.infof307.g01.model.gamehistory;

import java.util.Iterator;
import java.util.List;
import java.util.function.Consumer;

public class GameHistory implements Iterable<Game> {
    private final List<Game> games;

    public GameHistory(List<Game> games) {
        this.games = games;
        games.sort(Game::compareTo);
    }

    public int getNumberOfGames() {
        return games.size();
    }

    public int totalScore() {
        return games
                .stream()
                .mapToInt(Game::getScore)
                .sum();
    }

    @Override
    public Iterator<Game> iterator() {
        return games.iterator();
    }

    @Override
    public void forEach(Consumer<? super Game> action) {
        Iterable.super.forEach(action);
    }
}
